docs/advanced-data-preprocessing.md

```diff
@@ -184,6 +185,45 @@ We also allow users to pass a [`seed`](https://huggingface.co/docs/datasets/v3.2
 
 Note: If a user specifies data sampling they can expect the datasets to be mixed and individual samples in the dataset to not be broken unless the max_seq_len argument is smaller than the length of individual samples in the dataset
 
+### Dataset Splitting
+
+In addition to sampling and mixing, our library supports **dataset splitting**, which allows users to split a dataset into training and validation splits using the `split` field in the dataset config.
+
+This is especially useful when users want to split a single dataset (or multiple datasets) internally instead of supplying separate files for training and validation.
+
+#### How to Use
+
+To split a dataset, simply pro
+The `split` field in each dataset config allows users to internally divide a dataset into `train` and `validation` sets using fractional ratios.
+
+To use it, specify both `train` and `validation` values under the `split` key. Example:
+
+```yaml
+datasets:
+  - name: my_dataset
+    split:
+      train: 0.8
+      validation: 0.2
+    data_paths:
+      - "path/to/data.jsonl"
+```
+
+#### Split Support for Streaming vs Non-Streaming Datasets
+
+**Non-Streaming Datasets (`Dataset`, `DatasetDict`)**:
+- Supports arbitrary train/validation splits.
+- Both `train` and `validation` keys must be present under `split`.
+- The sum of `train + validation` must be in `(0, 1]`; less than 1.0 implies subset usage.
+- If no `split` is defined, the dataset is returned unchanged.
+
+**Streaming Datasets (`IterableDataset`, `IterableDatasetDict`)**:
+- Only supports full splits:
+  - Either `train: 1.0, validation: 0.0`
+  - Or `train: 0.0, validation: 1.0`
+- Partial splits like `train: 0.8, validation: 0.2` are not supported and will raise a `NotImplementedError`.
+- If no `split` is defined, the dataset is returned unchanged.
+- Streaming behavior must be explicitly enabled via `dataprocessor.streaming: true`.
+
```

Comment:
Add a line in bold that users need to explicitly set `eval_strategy` link to HF argument to something valid for eval to run and simply splitting a dataset will not result in eval and will likely result in an error. 

Also set a default config parameter in our training args `eval_strategy="no"`